File: src/main/java/com/cesello/observer/TabletClient.java
package com.cesello.observer;

public class TabletClient extends Observer{

    private Subject subject;

    public TabletClient(Subject subject) {
        this.subject = subject;
        subject.attach(this);
    }

    public void addMessage(String message) {
        this.subject.setState(message + " - sent from tablet.");
    }

    @Override
    void update() {
        System.out.println("Tablet stream: " + subject.getState());
    }
}

File: src/main/kotlin/com/cesello/observer/TabletClient.kt
package com.cesello.observer

class TabletClient(subject: Subject) : Observer(subject) {

    init {
        subject.attach(this)
    }

    fun addMessage(message: String) = subject.setState("$message - sent from tablet.")
    override fun update() = println("Tablet stream: ${subject.getState()}")
}